### ISSUE TITLE
Fine-tuning larger models

### ISSUE THREAD

**Issue by nmndeep** [tripletclip/TripletCLIP#3]
Hi,
Kudos on the great work.

I am trying to use TripletCLIP for fine-tuning a pre-trained CLIP ViT-L/14 model, I used 1M samples from your dataset and trained for 5-epochs equivalent at lr=5e-6 (did a small grid search over this), unfortunately I got not so good SugarCrepe(SC) numbers.

Specifically, for an ImageNet acc of 72\%: i get this following for SC 
`add_obj: 0.791
add_att: 0.769
replace_obj: 0.921
replace_att: 0.765
replace_rel: 0.626
swap_obj: 0.600
swap_att: 0.614
`

Could you point to an optimal training config that can be used for fine-tuning ViT-L/14 with TripletCLIP.
Also, it seems your code only works for a single-gpu setting: is there a possibility to extend this easily to multi-gpu setting(without affecting the hard-neg and hard-pos setup).

Best,
Naman

**Comment by Maitreyapatel** [comment]
Hi @nmndeep,
We only conducted the experiments on ViT-B/32 for pretraining due to resource constraints. While we followed the literature for finetuning, we only analyzed the ViT-B/32. So I am not able to share our findings on ViT-L/14.

Importantly, as reported in the paper, we do not recommend doing the fine-tuning experiments, as it limits the search space and could adversarially affect the model. Specifically, we found that the vision encoder is causing low compositional performance, and fine-tuning the pretrained model will not help much in the case of the hard negative images (like TripletData).

That said, sharing some comments that could help you:

> I got not so good SugarCrepe(SC) numbers.

Maybe a lower learning rate or fewer iterations may help. As mentioned above, changing the pretrained representations too much could degrade the performance. 


> Could you point to an optimal training config that can be used for fine-tuning ViT-L/14 with TripletCLIP.

We have not tasted this model earlier, but if time permits, I will do a quick fine-tuning and share the results/config.

 > Also, it seems your code only works for a single-GPU setting.

 We have adapted our code from open_clip, and it should be easy to transfer our loss function to open_clip. Mainly, the loss function needs to be slightly adjusted to gather tensors across the GPUs. Although this may require careful inspection on the dataloader. 
Tagging @Abhiramkns here if he can help add the multi-GPU support.

**Comment by nmndeep** [comment]
Hi, Thanks for the response. From my side, multi-GPU support is not a pressing thing, but might be good for your work in general.
You can close this from my side.
